### PR TITLE
ci: :construction_worker: add workflow to autogenerate images from pumls

### PR DESCRIPTION
**Pull request by signekb**
## Description 

So we don't have to do it manually (ever again 🌟 )

### CODE REVIEW

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
:tada: small change needed first :relaxed:

**Review by Aastedet — APPROVED** [review]

File: .github/workflows/puml-to-svg.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Generate SVG from PlantUML
+
+on:
+  workflow_call:
+    secrets:
+      github-token:
+        required: true
```
[the remark this answers is not in the document]
Comment (signekb) [reply .github/workflows/puml-to-svg.yaml:7]: Ah, ofc! I copy-pasted too hard 😛

File: .github/workflows/puml-to-svg.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Generate SVG from PlantUML
+
+on:
```
Comment (lwjohnst86) [inline .github/workflows/puml-to-svg.yaml:3]:
```suggestion
```

missed that!

File: .github/workflows/puml-to-svg.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      with:
+        fetch-depth: 1
+
+    - name: plantuml
+      id: plantuml
+      uses: grassedge/generate-plantuml-action@v1.5
+      with:
+        message: "chore: :bento: (re-)generate `.svg` from `.puml` files"
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (lwjohnst86) [inline .github/workflows/puml-to-svg.yaml:24]:
```suggestion
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

**Pull request merged**